src/components/TreeGrid.ts

```diff
@@ -46,13 +46,67 @@ const wrapContent =
     state: 'closed',
   });
 
+const hierarchyMapToRows = <T extends RowContents>(hierarchyMap: Map<number, T[]>, domainOptionRoot: T): Row<T>[] => {
+  const traverseHierarchy = (parent: T, depth: number, rows: Row<T>[]): Row<T>[] => {
+    const updatedRows: Row<T>[] = [
+      ...rows,
+      {
+        contents: parent,
+        depth,
+        isFetched: true,
+        state: 'open',
+      },
+    ];
+
+    // fetch all children of parent
+    const children: T[] | undefined = hierarchyMap.get(parent.id);
+    // if parent doesn't have children, return updatedRows
+    if (!children) {
+      return updatedRows;
+    }
+    children.forEach((child: T) => {
+      // if child has children, recursively call traverseHierarchy to get its branch
+      if (child.hasChildren) {
+        updatedRows.push(...traverseHierarchy(child, depth + 1, []));
+      }
+      // if child does not have children, add to Rows
+      else {
+        updatedRows.push({
+          contents: child,
+          depth: depth + 1,
+          isFetched: false,
+          state: 'closed',
+        });
+      }
+    });
+    return updatedRows;
+  };
+
+  // we use .get() to retrieve the values or children of domainOptionRoot
+  const domainOptionRootChildren: T[] | undefined = hierarchyMap.get(domainOptionRoot.id);
+
+  if (!domainOptionRootChildren) {
+    return [];
+  }
+
+  let rows: Row<T>[] = [];
+  const depth = 0;
+  domainOptionRootChildren.forEach((domainOptionRootChild) => {
+    rows = traverseHierarchy(domainOptionRootChild, depth, rows);
+  });
+
+  return rows;
+};
+
 type TreeGridProps<T extends RowContents> = {
   /** the columns to display */
   readonly columns: Column<T>[];
   /** the initial rows to display */
-  readonly initialRows: T[];
+  readonly initialHierarchy: Map<number, T[]>;
```

Comment:
The transformation layer looks great! Design question: Why does it exist within TreeGrid? Does this change affect the functionality of other usages of TreeGrid?

Reply:
Great question, I think we should tie this conversation with @s-rubenstein and @pshapiro4broad. I think TreeGrid.ts is mostly going to be used to visualize hierarchies for the exceptions of things like "Select concept sets". It's dangerous to make `readonly initialRows:  T[] | Map<number, T[]> ` because this frustrates TypeScript. Also, I think it's messy and quite ambiguous if we make both initialRows and initialHierarchy optional as: 
`readonly initialRows?: T[]`
`readonly initialHierarchy?: Map<number, T[]>`
After discussion with Phil, I think it's best to remove initialRows completely. I'd love to hear what others have to say, possibly a mobbing meeting?